src/shared-validator.cc

```diff
@@ -951,7 +952,13 @@ Result SharedValidator::OnRefFunc(const Location& loc, Var func_var) {
   Result result = CheckInstr(Opcode::RefFunc, loc);
   result |= CheckFuncIndex(func_var);
   if (Succeeded(result)) {
-    check_declared_funcs_.push_back(func_var);
+    // Reference declarations aren't needed for uses in global sections, and
+    // the use in a global itself counts as a declaration.
```

Comment:
Hoe about "References in initializer expressions are considered declarations, as opposed to references in function bodies that are considered usages".   (or something like that)